services/tables/src/main/java/com/linkedin/openhouse/tables/utils/TableUUIDGenerator.java

```diff
@@ -189,18 +188,26 @@ private Optional<UUID> extractUUIDFromRequestBody(
     if (snapshotStr == null) {
       return Optional.empty();
     }
+
+    // Extract tableLocation from table properties (openhouse.tableLocation)
+    // tableLocation should be the absolute path to the latest metadata file including scheme.
+    // Scheme is not present for HDFS and Local storages. See:
+    // https://github.com/linkedin/openhouse/issues/121
+    Storage storage =
+        storageManager.getStorageFromPath(
+            extractFromTblPropsIfExists(
+                tableURI,
+                snapshotsRequestBody.getCreateUpdateTableRequestBody().getTableProperties(),
+                TBL_LOC_RAW_KEY));
+    java.nio.file.Path databaseDirPath = Paths.get(storage.getClient().getRootPrefix(), databaseId);
     String manifestListKey = "manifest-list";
     java.nio.file.Path manifestListPath;
-    java.nio.file.Path databaseDirPath =
-        Paths.get(storageManager.getDefaultStorage().getClient().getRootPrefix(), databaseId);
-
     try {
-      manifestListPath =
-          Paths.get(
-              new Gson()
-                  .fromJson(snapshotStr, JsonObject.class)
-                  .get(manifestListKey)
-                  .getAsString());
+      String manifestListPathString =
+          new Gson().fromJson(snapshotStr, JsonObject.class).get(manifestListKey).getAsString();
+      manifestListPathString =
+          StringUtils.removeStart(manifestListPathString, storage.getClient().getEndpoint());
```

Comment:
is there a better way to achieve this without calling `storage.getClient().getEndpoint`, I'm ok with deferring this part to future pr.